app/components/InvalidNetworkModal/index.tsx

```diff
@@ -6,7 +6,7 @@ import { providers } from "ethers";
 import { Trans } from "@lingui/macro";
 
 interface Props {
-  provider: providers.JsonRpcProvider;
+  provider?: providers.JsonRpcProvider;
```

Comment:
Why do you allow for provider to be optional?

Reply:
The old hook used to return a fallback jsonrpcprovider when a user is not connected, the new one returns undefined.

This is better, because we can instantiate a provider anywhere in the code, there is no need or the hook to return one.